src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.Digest.cs

```diff
@@ -49,7 +49,10 @@ public static async Task<string> GetDigestTokenForCredential(NetworkCredential c
             string algorithm;
             if (digestResponse.Parameters.TryGetValue(Algorithm, out algorithm))
             {
-                if (algorithm != Sha256 && algorithm != Md5 && algorithm != Sha256Sess && algorithm != MD5Sess)
+                if (!algorithm.Equals(Sha256, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
Is there a guarantee that `algorithm != null` at this point?

Reply:
In general, TryGetValue can certainly succeed and still have `algorithm` be `null`, but I'm not sure if that can happen in practice here.

Reply:
Yes, the parser never returns `null` in the dictionary.